task2_api/task2_api.py

```diff
@@ -0,0 +1,102 @@
+from itertools import combinations
+import requests
+from collections import Counter
+import copy
+from datetime import datetime, timedelta
+from decimal import Decimal
+import csv
+
+
+class MovieData:
+    HEADERS = {
+        'accept': 'application/json',
+        'Authorization': 'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNkOTZjOSIsIn'
+                         'N1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.'
+                         'lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8'
+    }
+
+    BASE_URL = 'https://api.themoviedb.org/3/'
+
+    def __init__(self, num_pages):
+        self.num_pages = num_pages
+        self.movie_data = None
+        self.sorted_movie_data = None
+
+    def fetch_data(self, url):
+        response = requests.get(url, headers=self.HEADERS)
+        if response.status_code == 200:
+            return response.json()
+        print('Error fetching movie data.')
+        return None
+
+    def get_movie_data(self):
+        for i in range(1, self.num_pages):
+            url = f'{self.BASE_URL}discover/movie?include_adult=false&include_video=false' \
+                  f'&sort_by=popularity.desc&page={i}'
+
+            movie_data = self.fetch_data(url)
+            if movie_data is not None:
+                self.movie_data = movie_data['results']
```

Comment:
now you still have only one page at a time.
you just every iteration overwrite the  **self.movie_data**. so in the end you still will have only one page